open_city_profile/decorators.py

```diff
@@ -19,23 +19,45 @@ def wrapper(*args, **kwargs):
     return decorator
 
 
-def login_and_service_required(function):
-    """Decorator for checking that the user is logged in and service is known"""
+def _build_decorator(*test_funcs):
+    def decorator(function):
+        @wraps(function)
+        @context_helper(function)
```

Comment:
Now that the context_helper is used only here, I would maybe consolidate it and _build_decorator. Could be more easy to follow the flow.

Reply:
Good point.